### PR TITLE
Dispatch cursor position changes

### LINKED ISSUE #183: Dispatch the DescriptionEntered action when the cursor position changes
## Description
We need to dispatch this action whenever the cursor position changes, since that might change the current autocomplete suggestions
Looking into our Xamarin implementation is a good way of getting insight on all the problems that might arise from doing this manipulation, so I do recommend checking that battle hardened implementation instead of trying to come up with a naive solution that possible ignores corner cases. Needless to say, don't just blindly copy the Xamarin solution.
## Dependencies
Depends on #180 
## Definition of done
- [ ] The action is dispatched whenever the cursor position changes

### PR DESCRIPTION
**Pull request by heytherewill**
### Summary
This makes the description changed action be dispatched when the user changes the position of the cursor

### Why is it done like this and not some other way?
- There's no reliable way to listen to cursor position changes without using a sub-class
- The `requestFocus` method removes an ugly lambda from the activity
- We need the three constructors there because using `@JvmOverloads` (which creates a constructor that calls `super(context, null, 0)`) is not the same as calling `super(context)` for `EditText`s

### Relationships
Closes #183 

### Pull Request checklist
- [X] The code does not hinder the app's performance
- [X] The code does not introduce regressions
- [X] The code in this PR is tested thoroughly enough that changing the behavior to an incorrect one breaks tests

### Pointers
Type `LoggingReducer` in logcat and see the magic 

### Review template
- [ ] The code is thoroughly tested in well written and resilient manner
- [ ] The code is performant and leads to an optimal user experience
- [ ] The code is testable and does not cut corners for the sake of delivering
- [ ] The code considers older platforms well and will work in older devices just like it will in new ones

## Merge permissions
Anyone

### CODE REVIEW

File: timer/src/main/java/com/toggl/timer/startedit/ui/autocomplete/AutocompleteEditText.kt [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+    fun requestFocus(onFocusChanged: () -> Unit) {
+        setOnFocusChangeListener { _, _ ->
+            post { onFocusChanged() }
+            onFocusChangeListener = null
+        }
+        requestFocus()
+    }
+
+    fun setDescriptionChangedListeners(dispatch: () -> Unit) {
```
Comment (pdrj) [inline timer/src/main/java/com/toggl/timer/startedit/ui/autocomplete/AutocompleteEditText.kt:25]: Why the plural? It's always just one listener, right?

Comment (heytherewill) [reply timer/src/main/java/com/toggl/timer/startedit/ui/autocomplete/AutocompleteEditText.kt:25]: You set one for when the text changes and one for when the cursor changes

Comment (pdrj) [reply timer/src/main/java/com/toggl/timer/startedit/ui/autocomplete/AutocompleteEditText.kt:25]: OK, this is still the same listener though 😄 It just gets called in two places. I'm good with having that extra 's' in there then.

**Comment by pdrj** [comment]
> Besides the cosmetic comment, I've noticed that the listener would fire 3 times when you enter the edit fragment for a time entry that has description:
> Screenshot 2020-04-27 at 12 58 28
> 
> Are we fine with that?

What about this one @heytherewill?

File: timer/src/main/java/com/toggl/timer/startedit/ui/autocomplete/AutocompleteEditText.kt [hunk 2]
```diff
@@ -5,14 +5,30 @@ import android.text.TextWatcher
...
+@FlowPreview
+@ExperimentalCoroutinesApi
 class AutocompleteTextInputEditText : TextInputEditText {
     constructor(context: Context) : super(context)
     constructor(context: Context, attrs: AttributeSet) : super(context, attrs)
     constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int) : super(context, attrs, defStyleAttr)
 
-    private var dispatch: (() -> Unit)? = null
     private var descriptionChangeListener: TextWatcher? = null
+    // HACK: onSelectionChanged runs before the constructor, so this needs to be nullable
```
Comment (pdrj) [inline timer/src/main/java/com/toggl/timer/startedit/ui/autocomplete/AutocompleteEditText.kt:22]: :wtf:

Comment (heytherewill) [reply timer/src/main/java/com/toggl/timer/startedit/ui/autocomplete/AutocompleteEditText.kt:22]: I cry everytime 😭

**Pull request merged**